graphics/fonts/bdf.cpp

```diff
@@ -787,59 +797,76 @@ BdfFont *BdfFont::scaleFont(const BdfFont *src, int newSize) {
 		const BdfBoundingBox &box = data.boxes ? data.boxes[i] : data.defaultBox;
 		const BdfBoundingBox &srcBox = data.boxes ? src->_data.boxes[i] : src->_data.defaultBox;
 
+		int grayLevel = box.height * box.width / 3;
+		int dstPitch = (box.width + 7) / 8 ;
+
+#if DRAWDEBUG
+		int ccc = 'a';
+#endif
 		if (src->_data.bitmaps[i]) {
-			const int bytes = ((box.width + 7) / 8) * box.height; // Dimensions have been already corrected
+			int *dstGrayPointer = 0;
+			Graphics::Surface *srcSurf = scaledFont->scaleSingleGlyph(box.width, box.height, grayLevel, i + src->_data.firstCharacter, 
+																  	  box.xOffset, src->_data.height, srcBox.width, scale, &dstGrayPointer);
+			const int bytes = dstPitch * box.height;
+#if DRAWDEBUG
+			if (i == ccc) {
+				int *dstGray = *dstGrayPointer;
```

Comment:
What is the purpose of this declaration which immediately goes out of scope?

Reply:
dstGray is being used in the debug part of code, so needed declaration. It was getting its value from dstGrayPointer defined above, now changed it and removed dstGrayPointer, as it was redundant.

Reply:
If you want to declare it, then you declare it incorrectly. Try to compile with DRAWDEBUG on. It will not even compile!